docs/cli/logs.md

```diff
@@ -0,0 +1,50 @@
+# Function Logs
+
+OpenFaaS provides access to function logs via an API and the CLI: `faas-cli logs NAME`.
+
+This command will stream the logs for the named function.  By default, it will attempt to follow the logs, but you can control the behavior of the stream using these flags
+
+```sh
+--follow                 continue printing new logs until the end of the request, up to 30s (default true)
+--since duration         return logs newer than a relative duration like 5s
+--since-time timestamp   include logs since the given timestamp (RFC3339)
+--tail int               number of recent log lines file to display. Defaults to -1, unlimited if <=0 (default -1)
+```
+
+## Log structure
+The logs for a function will look
+
+```
+<RCF8601 Timestamp> <function name> (<container instance>) <msg>
+```
+
+where `msg` is the container logs, this typically contains stdout and stderr of the _contianer_.
+
+An example log output for `nodeinfo` from the function store is
+
+```sh
+2019-07-21 07:57:14.437219758 +0000 UTC nodeinfo (nodeinfo-867cc95845-p9882) 2019/07/21 07:57:14 Wrote 92 Bytes - Duration: 0.121959 seconds
+```
+
+## Filter and Search
+
+The CLI writes logs to stdout, so it can easily be chained with any of your favorite CLI tools: `grep`, `sed`, [`fzf`](https://github.com/junegunn/fzf) etc.
+
+For example, the Sentiment Analysis function in the Store prints a version string during startup.
+
+```sh
+faas-cli store deploy SentimentAnalysis
```

Comment:
What if you deploy with `--env write_debug`? Wouldn't that print the response? (could be a more useful example)

Reply:
I get the impression that you want something more _impressive_ in the logs.  Even with write_debug, the logs only look like 
```sh
2019-07-22 12:44:10.114247927 +0000 UTC sentimentanalysis (sentimentanalysis-7887c5d8c5-5rnb5) 2019/07/22 12:44:10 Duration: 0.541825 seconds
2019-07-22 12:44:11.426030609 +0000 UTC sentimentanalysis (sentimentanalysis-7887c5d8c5-5rnb5) 2019/07/22 12:44:11 Forking fprocess.
2019-07-22 12:44:11.426301995 +0000 UTC sentimentanalysis (sentimentanalysis-7887c5d8c5-5rnb5) 2019/07/22 12:44:11 Query
2019-07-22 12:44:11.426317507 +0000 UTC sentimentanalysis (sentimentanalysis-7887c5d8c5-5rnb5) 2019/07/22 12:44:11 Path  /
2019-07-22 12:44:11.882650374 +0000 UTC sentimentanalysis (sentimentanalysis-7887c5d8c5-5rnb5) 2019/07/22 12:44:11 Duration: 0.456617 seconds
2019-07-22 12:44:11.882708263 +0000 UTC sentimentanalysis (sentimentanalysis-7887c5d8c5-5rnb5) {"polarity": 0.0, "sentence_count": 1, "subjectivity": 0.0}
```

I guess I could grep on something like `sentence_count` instead, but I don't think it changes the example much

Reply:
Yes *this is impressive*, thank you 💪 

The bit I was looking for was the output from the function's code:

```
{"polarity": 0.0, "sentence_count": 1, "subjectivity": 0.0}
```

^ this bit.

Whenever I use the logs with a function, I am normally doing it to see an error or the output.

What do you think?

Reply:
👍 I updated it to grep on the `sentence_count`